src/api/user/entity.ts

```diff
@@ -42,6 +42,14 @@ export interface User {
   };
 }
 
+export interface SNSUser {
+  id: number,
+  nickname: string,
+  email: string,
+}
+
+export type Users = User | SNSUser;
```

Comment:
User**s** 면 아래처럼 추론되어야 할 것 같은데 의도랑 다른가요?
```suggestion
export type Users = Array<User | SNSUser>;
```

Reply:
단수형이라면 아래처럼 쓰는것도?
```suggestion
export type User = EmailUser | SNSUser;
```

Reply:
사용자는 EmailUser 또는 SNSUSer 둘 중 하나로만 선택되는 단수형이기 때문에 인터페이스명 변경하도록 하겠습니다.